### PR TITLE
Fix base64 encode/decode bug due to Pydantic issue

### PR DESCRIPTION
**Pull request by facutuesca**
Pydantic's `Base64Bytes` encodes bytes to base64 inserting newlines (`b'\n'`) every 76 characters. This is not correct for our use case, so this PR switches to our own `Base64Bytes` with a custom encoder.

See https://github.com/pydantic/pydantic/issues/9072#issuecomment-2361185555 for more context.

### CODE REVIEW

File: src/pypi_attestations/_impl.py [hunk 1]
```diff
@@ -34,6 +34,30 @@
...
+        """Encode bytes to base64."""
+        return base64.b64encode(value)
+
+    @classmethod
+    def decode(cls, value: bytes) -> bytes:
+        """Decode base64 bytes."""
+        return base64.b64decode(value, validate=True)
+
+
+Base64Bytes = Annotated[bytes, EncodedBytes(encoder=Base64EncoderSansNewline)]
```
Comment (woodruffw) [inline src/pypi_attestations/_impl.py:58]: Let's add a backstop test for this type as well, to assert that it absolutely never contains newlines in its encodings and fails when newlines are present while decoding.

**Review by woodruffw — APPROVED** [review]
Thanks @facutuesca! One comment + let's add a bugfix changelog entry for this 🙂

File: src/pypi_attestations/_impl.py (hunk 1 above)
Comment (facutuesca) [reply src/pypi_attestations/_impl.py:58]: done!

**Review by woodruffw — APPROVED** [review]
Thanks @facutuesca!

**Pull request merged**